src/main/java/org/assertj/assertions/generator/BaseAssertionGenerator.java

```diff
@@ -362,12 +401,20 @@ private String generateAssertionEntryPointMethodsFor(final Set<ClassDescription>
       String assertionEntryPointMethodContent = assertionEntryPointMethodTemplate.getContent();
       // resolve class assert (ex: PlayerAssert)
       // in case of inner classes like Movie.PublicCategory, class assert will be MoviePublicCategoryAssert
+      String customAssertionClass = fullyQualifiedAssertClassName(classDescription);
+      if (imports.contains(customAssertionClass)) {
```

Comment:
What about making the `generateAssertionEntryPointMethodsFor` return some wrapper over the content and the `imports`? We'll need duplicate logic if we handle in in 2 places.